packages/components/src/form/test/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Form from '../';
+
+// There is async code inside the callback handlers that we must flush
+// before asserting they were called.
+function flushPromises() {
+	return new Promise( function ( resolve ) {
+		setTimeout( resolve );
+	} );
+}
+
+describe( 'Form', () => {
+	it( 'should default to call the deprecated onSubmitCallback if it is provided.', async () => {
+		const mockOnSubmitCallback = jest.fn();
+		const mockOnSubmit = jest.fn();
```

Comment:
Personally, I prefer named mocks, instead of `mock*`convention.
```suggestion
		const onSubmitCallback = jest.fn().mockName('onSubmitCallback');
		const onSubmit = jest.fn().mockName('onSubmit');
```

So the assertion error would be 
```
expect( onSubmitCallback ).toHaveBeenCalledTimes( expected );

Expected number of calls: 1
Received number of calls: 0
```

Instead of 
```
expect( jest.fn() ).toHaveBeenCalledTimes( expected );

Expected number of calls: 1
Received number of calls: 0
…
…
await expect( mockOnSubmitCallback ).toHaveBeenCalledTimes( 1 );
```

As you don't need to parse (sometimes more complicated) assertion code by human eyes to get the clue.

Reply:
Thats a great idea, I'll fix that